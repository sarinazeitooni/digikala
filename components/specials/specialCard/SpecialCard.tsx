import React, {useEffect, useState} from 'react';
import style from './scss/special-card.module.scss';
import Progress from 'react-progressbar';
import NumberFormat from 'react-number-format';
import data from "../texts/data";
import AccessTimeOutlinedIcon from '@material-ui/icons/AccessTimeOutlined';

interface props {
    title: string,
    img: string,
    price: number,
    discount: number,
    digiPlus: string,
    sold: number
}

const SpecialCard = ({sold, title, img, price, discount, digiPlus}: props) => {
    const [oldPrice, setOldPrice] = useState(price);
    const [Discount, setDiscount] = useState(100 - discount);
    const [NewPrice, setNewPrice] = useState((oldPrice * Discount) / 100);
    return (
        <div className={style['card-container']}>
            <div style={{background: `url("${img}")`}} className={style['card-img']}/>
            <div className={style['card-title']}>{title}</div>
            <div className={style['card-digi-plus']}>
                {digiPlus !== "" ? <img className={style['digi-plus-icon']}
                                        src="https://www.digikala.com/static/files/3854bc8d.svg"/> : <span/>}
                {digiPlus}
            </div>
            <div className={style['prices-container']}>
                <div className={style['discount']}>{discount}%</div>
                <div className={style['old-price']}><NumberFormat value={price} displayType={'text'}
                                                                  thousandSeparator={true}/></div>
                <div className={style['new-price']}>
                    <NumberFormat value={NewPrice} displayType={'text'} thousandSeparator={true}/>
                    <span className={style['currency-price']}>{data.currency}</span>
                </div>
            </div>
            <div className={style['progress-bar-container']}>
                <div className={style['bar-container']}><Progress className={style['progress-bar']} color="#ef394e" height='4px' completed={sold}/></div>
                <div className={style['progress-text']}> <span>{sold}%</span> {data.sold}</div>
                <div className={style['limited-time']}><span> 22:21:00</span><AccessTimeOutlinedIcon/></div>
            </div>

        </div>
    )
}
export default SpecialCard;