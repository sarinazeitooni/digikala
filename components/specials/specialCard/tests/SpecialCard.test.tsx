import React from 'react'
import {shallow} from 'enzyme';
import SpecialCard from "../SpecialCard";
describe('SpecialCard component', () => {
    it('should render without throwing an error', ()=> {
        const wrapper = shallow(<SpecialCard/>);
        expect(wrapper).toMatchSnapshot();
    })
});