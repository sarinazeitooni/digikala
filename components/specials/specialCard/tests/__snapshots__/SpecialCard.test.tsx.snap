// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpecialCard component should render without throwing an error 1`] = `ShallowWrapper {}`;
